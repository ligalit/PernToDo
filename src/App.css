* {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
}

.section__app {
    min-width: 100%;
    min-height: 100%;
}

.container {
    max-width: 1200px;
    margin: 0 auto;
}

.section__todos {
    padding: 10px;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.section_todos-btns {
    width: 100%;
    display: flex;
    justify-content: center;
    margin-top: 10px;
    gap: 30px;
}

@media screen and (max-width: 480px) {
    .section_todos-btns {
        flex-direction: column;
        gap: 10px;
        margin: 20px 0;
    }
}

.section__todos-todo {
    padding: 15px;
    border: 2px solid teal;
    margin-top: 15px;
    display: flex;
    align-items: center;
    gap: 30px;
    justify-content: space-between;
}

