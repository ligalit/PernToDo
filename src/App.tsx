import React, {useMemo, useState} from 'react';
import './App.css';
import ToDoForm from "./components/ToDoForm";
import {IToDo} from "./interfaces/IToDo";
import ToDoList from "./components/ToDoList";
import Select from "./components/UI/Select";

function App() {
    const [toDo, setToDo] = useState<IToDo>({id: Date.now(), body: "", completed: false})
    const [toDos, setToDos] = useState<IToDo[]>([{id: 1, body: 'Clean Room', completed: false}]);
    const [filter, setFilter] = useState("all");

    const memoToDos = useMemo(() => {
        switch (filter) {
            case "active":
                return toDos.filter((t) => !t.completed);
            case "completed":
                return toDos.filter((t) => t.completed);
            default:
                return toDos;
        }
    }, [toDos, filter]);

    const handleAddToDo = () => {
        setToDos(prevState => [...prevState, toDo]);
        setToDo({id: Date.now(), body: "", completed: false});
    }

    const handleCompleteToDo = (id: number) => {
        setToDos(prevState => prevState.map((p) => p.id === id ? {...p, completed: !p.completed} : p));
    }

    const handleChangeFilter = (e: React.ChangeEvent<HTMLSelectElement>) => {
        setFilter(e.currentTarget.value);
    }

    const handleDeleteComplete = () => {
        setToDos(prevState => prevState.filter(p => !p.completed));
    };

    const select = (<Select data-testid="testSelect" onChange={handleChangeFilter}/>)

    return (
        <div className="section__app">
            <div className="container">
                <div className="section__todos">
                    <h1 data-testid="h1-todos" style={{textAlign: "center"}}>ToDos</h1>
                    <ToDoForm toDo={toDo} setToDo={setToDo} handleAddToDo={handleAddToDo}
                              handleDeleteComplete={handleDeleteComplete} select={select}/>
                    <ToDoList toDos={memoToDos} handleCompleteToDo={handleCompleteToDo}/>
                </div>
            </div>
        </div>
    );
}

export default App;
