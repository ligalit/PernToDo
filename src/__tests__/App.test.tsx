import React from 'react';
import App from '../App';
import {cleanup, render, screen} from "@testing-library/react";
import renderer from 'react-test-renderer';

afterEach(cleanup);

describe('App component', () => {
    test("render app elem", () => {
        const tree = renderer.create(<App/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("render h1 elem", () => {
        render(<App/>);
        const h1_todos = screen.getByTestId("h1-todos");
        expect(h1_todos).toBeInTheDocument();
    });
    test("render ToDoForm elem", () => {
        render(<App/>);
        const testToDoForm = screen.getByTestId("testToDoForm");
        expect(testToDoForm).toBeInTheDocument();
    });
    test("render ToDoList elem", () => {
        render(<App/>);
        const testToDoList = screen.getByTestId("testToDoList");
        expect(testToDoList).toBeInTheDocument();
    });
});
