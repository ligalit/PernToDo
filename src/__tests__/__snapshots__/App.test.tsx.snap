// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render app elem 1`] = `
<div
  className="section__app"
>
  <div
    className="container"
  >
    <div
      className="section__todos"
    >
      <h1
        data-testid="h1-todos"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        ToDos
      </h1>
      <form
        data-testid="testToDoForm"
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="What needs to be done?"
          value=""
        />
        <div
          className="section_todos-btns"
        >
          <button
            className="btn"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Add todos
          </button>
          <button
            className="btn"
            onClick={[Function]}
            style={
              Object {
                "border": "1px solid rgba(175,0,0,0.5)",
                "color": "rgba(175,0,0,0.5)",
              }
            }
            type="button"
          >
            Clear Completed
          </button>
          <select
            onChange={[Function]}
          >
            <option
              disabled={true}
              value=""
            >
              Choose from list
            </option>
            <option
              value="all"
            >
              All
            </option>
            <option
              value="active"
            >
              Active
            </option>
            <option
              value="completed"
            >
              Completed
            </option>
          </select>
        </div>
      </form>
      <div
        data-testid="testToDoList"
      >
        <div
          className="section__todos-todo"
        >
          <div
            style={
              Object {
                "textDecoration": "",
              }
            }
          >
            Clean Room
          </div>
          <div>
            <button
              className="btn"
              onClick={[Function]}
            >
              Complete To Do
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
